src/python/pants/python/python_setup.py

```diff
@@ -129,6 +136,16 @@ def register_options(cls, register):
                 "deprecation cycle."
             ),
         )
+        register(
+            "--invalid-lockfile-behavior",
+            advanced=True,
+            type=InvalidLockfileBehavior,
+            default=InvalidLockfileBehavior.warn,
```

Comment:
I think we talked about this a couple times, what the default should be. I can't remember what the conclusion was, but I'm thinking we should probably default to `error` for now, at least until we get feedback from users it's an annoying default. We want to encourage people to do the right thing, and warnings are too easy to miss (e.g. in CI).

Lockfiles are feature gated, so this is fine to change.

Reply:
Fine by me.